app/lib/actions/pulp/repository/presenters/puppet_presenter.rb

```diff
@@ -25,11 +25,14 @@ def humanized_details
           end
 
           def finished_count
-            task_result['finished_count']
+            return 0 unless task_progress_details.key?('modules')
+            task_progress_details['modules']['finished_count']
           end
 
           def total_count
-            task_progress_details['modules']['total_count'] || 0
+            return 0 unless task_progress_details.key?('modules')
+            return 0 if task_progress_details['modules']['total_count'].nil?
```

Comment:
It doesn't matter much to me, but it seems like you're doing the check to make sure task_progress_details is a hash with modules key, and then digging deeper into the hash, i think the two I gave I've seen more in other ruby code